src/vaadin-text-field-material-outlined.ts

```diff
@@ -0,0 +1,242 @@
+      }
+
+      [part="label"] {
+        transition: transform 0.175s, color 0.175s, width 0.175s;
+        transition-timing-function: ease, ease, step-end;
+        left: 8px;
+        top: 14px;
+        padding: 0 8px;
+        background-color: #fff;
+        z-index: 2;
+      }
+/*
+      :host([focused]:not([invalid])) [part="label"] {
+        background-color: #fff;
+        z-index: 2;
+      } */
+
+      /* TODO: using unsupported selector to fix IE11 (even thought the label element is scaled down,
+         the 133% width still takes the same space as an unscaled element */
+      ::-ms-backdrop,
+      .vaadin-text-field-container {
+        overflow: hidden;
+      }
+
+      /* Same fix for MS Edge ^^   */
+      @supports (-ms-ime-align:auto) {
+        .vaadin-text-field-container {
+          overflow: hidden;
+        }
+      }
+
+      :host(:hover:not([readonly]):not([invalid])) [part="input-field"]::before {
+        opacity: var(--_material-text-field-input-line-hover-opacity, 0.87);
+      }
+
+      :host([focused]:not([invalid])) [part="label"] {
+        color: var(--material-primary-text-color);
+      }
+
+      :host([focused]) [part="input-field"]::after,
+      :host([invalid]) [part="input-field"]::after {
+        opacity: 1;
+        transform: none;
+        transition: transform 0.175s, opacity 0.175s;
+      }
+
+      :host([invalid]) [part="input-field"]::after {
+        background-color: var(--material-error-color);
+      }
+
+      :host([input-prevented]) [part="input-field"] {
+        color: var(--material-error-text-color);
+      }
+
+      :host([disabled]) {
+        pointer-events: none;
+      }
+
+      :host([disabled]) [part="input-field"] {
+        color: var(--material-disabled-text-color);
+      }
+
+      :host([disabled]) [part="input-field"]::before {
+        background-color: transparent;
+        background-image: linear-gradient(90deg, var(--_material-text-field-input-line-background-color, #000) 0, var(--_material-text-field-input-line-background-color, #000) 2px, transparent 2px);
+        background-size: 4px 1px;
+        background-repeat: repeat-x;
+      }
+
+      /* Only target the visible floating label */
+      :host([has-label]:not([has-value]):not([focused]):not([invalid]):not([theme~="always-float-label"])) [part="label"] {
+        /* IE11 doesn’t work with calc inside the translate function, so we need to have a fixed pixel value instead of 50% + 16px */
+        transform: scale(1) translateY(32px);
+        transition-timing-function: ease, ease, step-start;
+        pointer-events: none;
+        left: 8px;
+        transition-delay: 0.1s;
+      }
+
+      /* Slotted content */
+
+      [part="input-field"] ::slotted(*:not([part="value"]):not([part\$="-button"]):not(input):not(textarea)) {
+        color: var(--material-secondary-text-color);
+      }
+
+      [part="clear-button"]::before {
+        content: var(--material-icons-clear);
+      }
```

Comment:
Can we account for prefixes and suffixes? They get cut off if applied: `<vaadin-text-field placeholder="Username" label="Username">
            <div slot="suffix">$</div>
        </vaadin-text-field>`
and the prefix goes outside the border according to https://storage.googleapis.com/spec-host-backup/mio-design%2Fassets%2F1Kezf2nmTVbQdOfNajrotDxeIKs1WlwHl%2Ftextfields-type-stroke.png

We should add it to the backlog if it's difficult since Kasey really wants them NOW